### ISSUE TITLE
OpenGL backend for All Platform Windows Linux MacOS IOS Android etc

### ISSUE THREAD

**Issue by insinfo** [AvaloniaUI/Avalonia#985]
It would be fantastic if a hardware-accelerated backend was implemented, bringing the graphics APIs of each platform equal to or higher

**Comment by insinfo** [comment]
The problem with Avalonia is that it is not written on top of a low-level API like OpenGL/Direct3D/Vulkan/Metal, Avalonia is a lot of overhead, which is terrible for
a graphical framework/library

**Comment by kekekeks** [comment]
Skia can render using OpenGL and we had support for that in our C++-based backend. It's planned to reimplement this with Skia# and in managed

**Comment by insinfo** [comment]
See this very interesting material on 2D rendering, and interesting demonstration on the performance of Skia

https://www.youtube.com/watch?v=bCrohG6PJQE
https://www.youtube.com/watch?v=zI8cJMjEVBE
https://www.youtube.com/watch?v=vE0ZzqK1_dE
https://www.youtube.com/watch?v=1yQkAn1w-cU
http://web.engr.oregonstate.edu/~mjb/cs519/Projects/Papers/AcceleratingVectorGraphicsRendering.pdf
http://developer.download.nvidia.com/devzone/devcenter/gamegraphics/files/opengl/gpupathrender.pdf
http://on-demand.gputechconf.com/gtc/2012/presentations/S0024-GPU-Accelerated-Path-Rendering.pdf

**Comment by kekekeks** [comment]
@insinfo the thing is that you want us to implement our own 2D rendering engine in C#. Such engine would be indeed very useful, but it's not particularly tied or related to UI framework itself and should live in a separate library (just like XAML parsers do). It also will require tremendous amount of work and an expert in that field.

TL;DR: It there would be a 2D hardware renderer implementation that's better than Skia and meets our needs in terms of API, then, for sure, we'll make the effort for integrating it. Our own implementation? Probably, no. We lack funds and expertise to do that in foreseeable future.

**Comment by kekekeks** [comment]
BTW, thanks for these links, very interesting to read :+1:

**Comment by insinfo** [comment]
@kekekeks
Thank you for your understanding and attention and I understand the difficulties and complexities in this area. When I have a free time I will try to contribute to the project

**Comment by juepiezhongren** [comment]
@kekekeks have u consider to use skiaSharp for control rendering?

**Comment by maxkatz6** [comment]
We do use opengl backend for a long time now
